qiskit/circuit/quantumcircuit.py

```diff
@@ -1737,16 +1740,24 @@ def qasm(
             elif operation.name == "barrier":
                 qargs = ",".join(bit_labels[q] for q in instruction.qubits)
                 instruction_qasm = "barrier;" if not qargs else f"barrier {qargs};"
+            elif hasattr(operation, "has_qasm2") and not operation.has_qasm2():
+                raise CircuitError(
+                    f"No OpenQASM 2 representation is defined for operation {operation.name}"
+                )
```

Comment:
`operation.has_qasm2`? We shouldn't be adding new magic interfaces; we're trying to get rid of them.

Reply:
I'll try to find another way to do this.

Reply:
Removed in 68c1832